File: packages/gpui/workstation/src/persistence.rs
use crate::types::WorkspaceLayoutSnapshot;

pub fn serialize_workspace_layout_snapshot(
    snapshot: &WorkspaceLayoutSnapshot,
) -> Result<String, serde_json::Error> {
    serde_json::to_string_pretty(snapshot)
}

pub fn parse_workspace_layout_snapshot(
    serialized: &str,
) -> Result<WorkspaceLayoutSnapshot, serde_json::Error> {
    serde_json::from_str(serialized)
}

File: packages/svelte/components/src/persistence.ts
import type { WorkspaceLayoutSnapshot } from "./types";

export function serializeWorkspaceLayoutSnapshot(snapshot: WorkspaceLayoutSnapshot): string {
  return JSON.stringify(snapshot, null, 2);
}

export function parseWorkspaceLayoutSnapshot(serialized: string): WorkspaceLayoutSnapshot {
  return JSON.parse(serialized) as WorkspaceLayoutSnapshot;
}
